### DISCUSSION TITLE
Goodreads scrubber

### DISCUSSION THREAD

**Discussion by yotsimitsu** [seanap/Plex-Audiobook-Guide#95] [Ideas]
Hi,
first thanks for the great work, this really helped me to bet my library in order. 

I have one thing, would you be able to create one more source for GoodReads? There used to be one, but it stopped working recently, probably some API change and without it it's very hard to tag old audiobooks that aren't on audible.

Thank you

**Comment by seanap** [comment]
Hey, I didn't develop the Goodreads scraper but I took a brief look and it looks like the majority of tags now throw an error and will need to be reconfigured.  This isn't particularly hard but it'll take time to search the new html for the new locations of the data we need.  If time allows I can take a closer look, but hopefully the original author will update it.

**Reply by yotsimitsu** [reply]
I expected as much... I already left a comment in the origibal thread but I don't think the original maker is still active.

Sure, thank you! I know it would take more of your time but I also feel like goodreads is a good source to have since audible is missing a lot of books.

Appreciate your answer!

**Reply by seanap** [reply]
Hey this may be a lot easier because the changes to goodreads site clearly list and label the metadata we're after.  Couple questions:

1. Was the old script pulling the poster size (tall rectangle) cover? or was it somehow getting the square cover?
2. I don't see any provisions for pulling the narrator, how were you doing that?
3.  Is it worth pulling the publisher tag since goodreads search returns the book not the audiobook?  (ex. The Martian book was through Crown, the audiobook was through Podium).  I really don't like tagging incorrect data so I'm considering just not pulling this.

**Reply by yotsimitsu** [reply]
Nice!

1. It was pulling the rectangle, I don't think they have a square alternative. 
2. No narrator unfortunately, goodreads just doesn't have that kind of data.
3. You can ignore the publisher I think, it would be the book publisher and not the audiobook.

**Reply by seanap** [reply]
Not as easy as I thought it was going to be.  Here's a bare bones version that should grab the Cover, Author, Title, Summary, and Year. https://pastebin.com/LgLpWjbM

**Reply by yotsimitsu** [reply]
You are the best! I'm sure there will be others that will also appreciate this. Anybody with a big library that has some more obscure audiobooks will benefit from this.

**Reply by bcleonard** [reply]
It looks this is broken again (at least for me).   Has anybody else experienced this?
